### PR TITLE
Taste

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
add: Nanotrasen reminds its employees that they have ALWAYS been able to taste. Anyone claiming that they've recently only just gained the ability to taste are probably Syndicate agents.
/:cl:

- Mobs can now taste things! Currently only carbons and corgis can taste things.
- Taste sensitivity (how good you are at detecting minute flavours) are based on your tongue. Lizards are slightly better than humans at tasting things, while abductors and skeletons have no sense of taste at all.
- A wide number of reagents share similar or the same taste description, nearly all medicines and toxins taste of "bitterness", and the only REALLY unique flavours are limited to alcoholic beverages.
- Some reagents have stronger flavour than others, and can drown out weaker flavours.
- Admin only tool, the "taster", a tongue you hit things with, and taste their `reagents` var. Yes, technically, I suppose  you could taste humans, but only the reagents inside them.

- Flavours of various food things are a work in progress, because I don't know what's in half of our recipes. Since food inherits the taste of its ingredients, properly crafted food will have a different taste from spawned food.

Why? Because it's TASTING THINGS, it's cool, makes the world slightly more real, makes chefs and bartenders a little happy, and means suicidal assistants can try to ID random beakers of chemicals they find.

### CODE REVIEW

**Comment by Cheridan** [comment]
cherry flavored lube, my favorite!

**Comment by Armhulen** [comment]
CAWFEE CHEETOS CHICKEN

File: code/modules/food_and_drinks/food/snacks.dm [hunk 1]
```diff
@@ -16,11 +16,24 @@
...
 
 	//Placeholder for effect that trigger on eating that aren't tied to reagents.
 
+/obj/item/weapon/reagent_containers/food/snacks/add_initial_reagents()
+	if(tastes && tastes.len)
+		if(list_reagents)
+			for(var/rid in list_reagents)
+				var/amount = list_reagents[rid]
+				if(rid == "nutriment" || rid == "vitamin")
+					reagents.add_reagent(rid, amount, tastes.Copy())
```
Comment (Iamgoofball) [inline code/modules/food_and_drinks/food/snacks.dm]: i really dont know how i feel about adding even more list processing to reagent transfers

Comment (coiax) [inline code/modules/food_and_drinks/food/snacks.dm]: that doesn't even make sense for the code you're highlighting

**Comment by bgobandit** [comment]
please make lube cherry flavored

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 2]
```diff
@@ -176,6 +191,7 @@
 	id = "soymilk"
 	description = "An opaque white liquid made from soybeans."
 	color = "#DFDFC7" // rgb: 223, 223, 199
+	taste_description = "soy milk"
```
Comment (lordpidey) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm]: There's gotta be a more humorous description we can put here.  How about "Milk and lies"

Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm]: "it tastes like mil.... wait nevermind, that certainly isn't the milk i'm used to!"

**Comment by lordpidey** [comment]
Could you add a flavor to saline glucose?

Oh, and in spray tan's effect has a quip about how it tastes if you ingest it.  Might want to change that.

**Comment by Repukan** [comment]
>Yes, technically, I suppose you could taste humans, but only the reagents inside them.

So uh, what does a liar taste like?

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm (hunk 2 above)
Comment (Anonmare) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm]: Tastes like betrayal

**Comment by Anonmare** [comment]
@Reoukan Like an ahelp

**Comment by PKPenguin321** [comment]
Allow us to set our flavor in the character setup thing so if the chef makes us into burgers we can still be ~unique~

**Comment by PKPenguin321** [comment]
/~unique/~

**Comment by PKPenguin321** [comment]
Fuck

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 3]
```diff
@@ -80,27 +114,32 @@
...
 	color = "#899613" // rgb: 137, 150, 19
+	taste_description = "watery milk"
 
 /datum/reagent/consumable/soysauce
 	name = "Soysauce"
 	id = "soysauce"
 	description = "A salty sauce made from the soy plant."
 	nutriment_factor = 2 * REAGENTS_METABOLISM
 	color = "#792300" // rgb: 121, 35, 0
+	taste_description = "unami"
```
Comment (erwgd) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: *umami

File: code/modules/mob/living/taste.dm [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+/mob/living/carbon/get_taste_sensitivity()
+	var/obj/item/organ/tongue/tongue = getorganslot("tongue")
+	if(istype(tongue))
+		. = tongue.taste_sensitivity
+	else
+		. = 0 // can't taste anything without a tongue
+
+// non destructively tastes a reagent container
+/mob/living/proc/taste(var/datum/reagents/from)
```
Comment (RemieRichards) [inline code/modules/mob/living/taste.dm]: no ```var/``` in proc arguments

File: code/modules/mob/living/taste.dm [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+	if(last_taste_time + 50 < world.time)
+		var/datum/reagents/temp = new(1)
```
Comment (RemieRichards) [inline code/modules/mob/living/taste.dm]: ...why?

**Comment by CakeQ** [comment]
Yes please

**Comment by CakeQ** [comment]
@coiax Any plans on adding in the extended bar system as well?

File: code/modules/reagents/chemistry/reagents/food_reagents.dm (hunk 3 above)
Comment (Iamgoofball) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: :+1: what he said I'm a culinary nut so I'm triggered rn

**Comment by coiax** [comment]
@CakeQ I don't know what that is.

**Comment by coiax** [comment]
@lordpidey Spray tan already tastes like sour oranges.

**Comment by Dorsisdwarf** [comment]
''Lizards are slightly better than humans at tasting things''

Unfortunately our design rules say lizards arent allowed to be better than humans, out we go.





👍

**Comment by Anonmare** [comment]
@Dorsisdwarf Lizards can't taste anything poisonous when?

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 6]
```diff
@@ -86,18 +93,21 @@
...
 	description = "Delicious juice made from watermelon."
 	color = "#863333" // rgb: 134, 51, 51
+	taste_description = "juicy watermelon"
 
 /datum/reagent/consumable/lemonjuice
 	name = "Lemon Juice"
 	id = "lemonjuice"
 	description = "This juice is VERY sour."
 	color = "#863333" // rgb: 175, 175, 0
+	taste_description = "sourness"
```
Comment (ma44) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm]: Should taste like dosh **(I'm sorry)**

Comment (kevinz000) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm]: ban he

**Comment by tristan1333** [comment]
https://www.youtube.com/watch?v=SLP9mbCuhJc

**Comment by XDTM** [comment]
Is there a simple-ish way of making mobs lose taste? It's in a viro symptom description, it would be cool to actually implement it.

**Comment by coiax** [comment]
I can easily hack it, but I think it would be better with an organ damage system, where a damaged tongue loses its tasting (and also anti-lisp) ability.

File: code/modules/food_and_drinks/food/snacks_soup.dm [hunk 7]
```diff
@@ -51,79 +58,90 @@
...
 	bonus_reagents = list("nutriment" = 1, "tomatojuice" = 10, "vitamin" = 3)
 	list_reagents = list("nutriment" = 5, "tomatojuice" = 10, "vitamin" = 3)
+	tastes = list("tomato" = 1)
 
 /obj/item/weapon/reagent_containers/food/snacks/soup/milo
 	name = "milosoup"
 	desc = "The universes best soup! Yum!!!"
 	icon_state = "milosoup"
 	bonus_reagents = list("nutriment" = 1, "vitamin" = 3)
+	tastes = list("milo" = 1) // wtf is milo
```
Comment (Nabski89) [inline code/modules/food_and_drinks/food/snacks_soup.dm]: In this case it's a tofu soup.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 8]
```diff
@@ -25,44 +27,76 @@
...
-	color = "#664330" // rgb: 102, 67, 48
 
-/datum/reagent/consumable/vitamin/on_mob_life(mob/living/M)
-	if(prob(50))
-		M.heal_bodypart_damage(1,1, 0)
-		. = 1
+	brute_heal = 1
+	burn_heal = 1
+	blood_gain = 0.5
+
```
Comment (Nabski89) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: Did this change from a 50% chance to heal to 100%?

**Pull request merged**